Landscape

Nyefari opened the pull request:
PR to move Landscape generation into the master for MVP 1 (Also contains a version of Rad's Camera which allows panning and zoom) Resolves #1 

Landscape generation works by taking the height data from a grayscale png and generating a land mesh (generation of mesh is part of the land mod). 

Currently only creates one "tile" of land. I plan to implement it in the future to create/stream them based on camera(s?) position, maybe with LOD considerations for land meshes that are farther from the camera?

Known Issues:
1. Default Bevy render pipeline limits verteex indices to Uint16s, so meshes can have a maximum of 65k vertexes. This functionally limits height map tiles to 256 x 256

2. Does not recalculate normals in any way shape or form.

ocornoc committed:
Added new camera, added test

ocornoc committed:
Make CameraBPAction pub, weight

ocornoc committed:
Move camera movement into camtest

ocornoc committed:
Renamed camera test

ocornoc committed:
Fix signal deduping

ocornoc committed:
Renamed the camera test again

ocornoc committed:
Fix typo signalling -> signaling

ocornoc committed:
Rename "cams" to "acts"

ocornoc committed:
Added zooming

ocornoc committed:
Add distance scaling, cameras now have individual config

ocornoc committed:
Camera locking is now camera-specific

ocornoc committed:
Fix typo in cam_zoom_winedge_test docs

Nyefari committed:
Test Commit please ignore. (This is so I can pull from the code when I might go back to some of these idioms)

ocornoc committed:
New continuous panning test

ocornoc committed:
Added newline in zoom for consistency

ocornoc committed:
fixed NaN bug in plane trans

ocornoc committed:
Add .vscode to gitignore

Nyefari committed:
Add basic plane mesh, color and deform based off of position wrt origin using shaders. Spawn said meshes w/ shaders into world to fill an area of the world.

ocornoc committed:
Moved tests to examples

Nyefari committed:
Final Vers of MVP1 Land. 2 issues (See Github)

Mess with file sizes to figure out exact limits

Nyefari committed:
no message

Nyefari committed:
Mess with file sizes to figure out exact limits

Nyefari committed:
Revert "Mess with file sizes to figure out exact limits"

This reverts commit 05e85ba3022caa15547b41fc193c06e09a0bdcad.

Nyefari committed:
Revert "Revert "Mess with file sizes to figure out exact limits""

This reverts commit cbccca834aef809f2ea9e57fb15437e7ce83a917.

Nyefari committed:
Final Vers of MVP1 Land. 2 issues (See Github)

Nyefari committed:
Merge branch 'Landscape' of https://github.com/CylonSB/bounded-planet into Landscape

martindevans commented on .vscode/launch.json:
```diff
@@ -0,0 +1,138 @@
+{
```
Presumably this file should not be included? I assume not since you have specifically ignored `.vscode` which this is in.

martindevans commented on src/land/land.rs:
```diff
@@ -0,0 +1,72 @@
+) -> bevy::prelude::Mesh {
+
+    //gets a reference to the data needed
+    let land_texture = textures.get(&land_texture_handle).unwrap();
```
Use `expect("error message")` in place of `unwrap`. Even better, make the function return `Result<bevy::prelude::Mesh, SomeErrorType>` and don't panic at all.

martindevans commented on src/media/shaders/land.frag:
```diff
@@ -0,0 +1,11 @@
+#version 450
```
is this shader still needed? as far as I can see it's no longer used.

martindevans commented on src/media/shaders/land.vert:
```diff
@@ -0,0 +1,29 @@
+#version 450
```
is this shader still needed? as far as I can see it's no longer used.

martindevans reviewed (changes requested):
We also need a discussion in Discord re. comitting assets. These ones are tiny so it doesn't matter, but if we start comitting all game assets it's going to bloat the repo a lot.

Nyefari committed:
Make changes from Martin's Code review:
1. Fixed vscode file issue
2. Handle errors more correctly for implemented functions.
3. Added test_assets folder and added it to .gitignore . Will use in the future for testing assets.

Nyefari committed:
Merge branch 'master' into Landscape
